code/game/objects/items/weapons/twohanded.dm

```diff
@@ -405,6 +406,32 @@
 	if(wielded)
 		. = ..()
 
+
+/obj/item/weapon/twohanded/required/chainsaw/doomslayer
+	name = "OOOH BABY"
+	desc = "<span class='warn'>VRRRRRRR!!!</span>"
+	force = 20
+	armour_penetration = 100
+
+/obj/item/weapon/twohanded/required/chainsaw/doomslayer/attack_self(mob/user)
+	on = !on
+	user << "You viciously yank the cord dangling from \the [src], [on ? "and it roars to life, screaming along with your bloodlust." : "grows dormant."]"
```

Comment:
make these parts of the ternary (```?```) operation variables, then you don't have to copy paste this entire proc.
eg:
```
var/chainsaw_on_fluff = "vroom it is now on"
var/chainsaw_off_fluff = "vroom it's off now"
```